katello/katello/hostname-change.rb

```diff
@@ -256,8 +256,24 @@ def run
 
       public_dir = "/var/www/html/pub"
       public_backup_dir = "#{public_dir}/#{@old_hostname}-#{self.timestamp}.backup"
-      STDOUT.puts "deleting old certs"
+      
+      STDOUT.puts "removing old cert rpms"
+      cert_packages = [
+        "foreman-proxy-client*",
+        "qpid-client-cert*",
+        "qpid-router-server*",
+        "tomcat*",
+        "apache*",
+        "qpid-broker*",
+        "puppet-client*",
+        "foreman-proxy*",
+        "qpid-router-client*",
+        "foreman-client*"
+      ]
+      cert_rpms = cert_packages.map { |pkg| "#{@old_hostname}-#{pkg}*" }.join(' ')
+      self.run_cmd("yum remove -y #{cert_rpms}")
 
+      STDOUT.puts "deleting old certs"
```

Comment:
I like to put the action being taken above the action itself, so if anything goes wrong, its easy to see at what step it happens.